src/com/zone5cloud/retrofit/core/OkHttpClientInterceptor_Authorization.java

```diff
@@ -195,33 +202,44 @@ private void refreshIfRequired(Chain chain, String baseUrl) {
 				token = this.token.get();
 				if (token != null && token.isExpired()) {
 					try {
-						if (token.getRefreshToken() != null) {
-							// do a Cognito refresh
-							String username = token.extractUsername();
-							if (username != null) {
-								RequestBody body = new FormBody.Builder().add("client_id", clientID)
-																		 .add("client_secret", clientSecret)
-																		 .add("grant_type", GrantType.REFRESH_TOKEN.toString())
-																		 .add("username", username)
-																		 .add("refresh_token", token.getRefreshToken()).build();
-								String url = baseUrl + Users.NEW_ACCESS_TOKEN;
-								Request authRequest = new Request.Builder().url(url).post(body).build();
+							if (token.getRefreshToken() != null) {
+								// do a Cognito refresh
+								String username = token.extractUsername();
+//
+								if (username != null) {
+									// Add Api-Key and Api-Key-Secret to body if the request url matches zone5BaseUrl
+									if(originalRequestUrl.contains(zone5BaseUrl)) {
```

Comment:
the refresh is ALWAYS to the zone5 server. ALWAYS add these headers.

So the original request coming in might be to the SpecializedBRS server, but the cognito token is expired so we need to send a refresh token request to the Zone5 server at this point, before we send the SpecializedBRS request.